### PR TITLE
DAOS-12222 pool: more INFO logs; test: DEBUG in create/destroy

### PR DESCRIPTION
**Pull request by kccain**
Change pool destroy code so some essential log entries are
D_INFO (rather than D_DEBUG). This is done so when pool destroy fails
(or hangs), a course-grained understanding of progress is visible
before running a reproducer with log_mask: DEBUG. Additionally,
some D_ERROR entries are added so slightly more information is
reported about the location of a failed destroy.

Also with this change, the functional test code uses the
dmg server set-logmasks command to elevate the engine log_mask
to "DEBUG" immediately preceding any pool create or pool destroy.
Correspondingly, immediately following, it executes
dmg server set-logmasks (no argument) to restore the original
log_mask. This is done to lessen the need for test reproducer
runs when a test failed in pool create/destroy (fairly common)
but whose engine configuration is log_mask: INFO (also common).
Additional raise_exception boolean argument is provided in various
methods to override the exit_status_exception class attribute for
specific command invocations such as set-logmasks that will
fail if, for example, an engine has been stopped (commonly done
by tests).

Test-tag: pr daily_regression

Required-githooks: true

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/command_utils.py [hunk 1]
```diff
@@ -149,17 +159,22 @@ def _run_process(self):
         # Clear any previous run results
         self.result = None
         command = str(self)
+        if raise_exception is not None:
+            do_raise = raise_exception
+        else:
+            do_raise = self.exit_status_exception
```
Comment (phender) [inline src/tests/ftest/util/command_utils.py:165]: We could save ourselves an extra variable here:
```suggestion
        if raise_exception is None:
            raise_exception = self.exit_status_exception
```

File: src/tests/ftest/util/command_utils.py [hunk 2]
```diff
@@ -149,17 +159,22 @@ def _run_process(self):
...
+
         try:
             # Block until the command is complete or times out
             self.result = run_command(
-                command, self.timeout, self.verbose, self.exit_status_exception,
+                command, self.timeout, self.verbose, do_raise,
```
Comment (phender) [inline src/tests/ftest/util/command_utils.py:170]: Then, to follow on from the previous comment:
```suggestion
                command, self.timeout, self.verbose, raise_exception,
```

File: src/tests/ftest/util/command_utils.py [hunk 3]
```diff
@@ -149,17 +159,22 @@ def _run_process(self):
...
                 self.output_check, env=self.env)
 
         except DaosTestError as error:
             # Command failed or possibly timed out
             raise CommandFailure(str(error))    # pylint: disable=raise-missing-from
 
-        if self.exit_status_exception and not self.check_results():
+        if do_raise and not self.check_results():
```
Comment (phender) [inline src/tests/ftest/util/command_utils.py:177]: And to follow on from the previous two comments:
```suggestion
        if raise_exception and not self.check_results():
```

File: src/tests/ftest/util/command_utils.py [hunk 4]
```diff
@@ -527,14 +547,14 @@ def run(self):
...
-            super().run()
+            super().run(raise_exception)
         except CommandFailure as error:
             raise CommandFailure(
                 "<{}> command failed: {}".format(
                     self.command, error)) from error
         return self.result
 
-    def _get_result(self, sub_command_list=None, **kwargs):
+    def _get_result(self, sub_command_list=None, raise_exception=None, **kwargs):
```
Comment (phender) [inline src/tests/ftest/util/command_utils.py:557]: This function is missing one more edit.  On https://github.com/daos-stack/daos/blob/53e17d9b6eadee99707ce2d9266d15ceb78c67fe/src/tests/ftest/util/command_utils.py#L597 the line should be modified to be:
```
        return self.run(raise_exception)
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/command_utils.py [hunk 5]
```diff
@@ -571,17 +593,24 @@ def _get_result(self, sub_command_list=None, **kwargs):
...
             json_err (bool, optional): If True, return JSON
                 even if the command fails.
+            raise_exception (bool, optional): whether or not to raise an exception if the command
+                fails. This overrides the self.exit_status_exception
+                setting if defined. Defaults to None.
             kwargs (dict): Parameters for the command.
         """
+
+        self.log.info("CommandWithSubCommand._get_json_result: kwargs=%s", kwargs)
+
```
Comment (phender) [inline src/tests/ftest/util/command_utils.py:613]: I realize this is still a draft, but presumably this is debug code and will be removed, right?

**Review by liw — APPROVED** [review]
Thanks.

**Review by phender — APPROVED** [review]

File: src/tests/ftest/util/command_utils.py [hunk 6]
```diff
@@ -124,21 +124,31 @@ def with_exports(self):
         """
         return " ".join([self.env.to_export_str(), str(self)]).strip()
 
-    def run(self):
+    def run(self, raise_exception=None):
```
Comment (ashleypittman) [inline src/tests/ftest/util/command_utils.py:127]: This created a bunch of pylint errors for classes that override this method due to different parameter names and whilst pylint does pick it up we didn't gate the PR on it as we only fail PRs for failures in the modified code, not failures that are created elsewhere.

Comment (kccain) [reply src/tests/ftest/util/command_utils.py:127]: Running pylint locally on the commit preceding this one, and again on this commit seems to show the following pylint issue was added by the commit of this PR
src/tests/ftest/util/fio_utils.py:172:12: E1121: Too many positional arguments for method call (too-many-function-args)

**Pull request merged**